packages/pn-commons/src/services/message.service.ts

```diff
@@ -6,7 +6,7 @@ import _ from 'lodash';
  * @param  {{response:{status:number;};}} error
  * @returns IAppMessage
  */
-export const createAppError = (error: { response: { status: number } }): IAppMessage => {
+export const createAppError = (error: { response: { status: number; data: { title: string} } }): IAppMessage => {
```

Comment:
Il message service è qualcosa che dovrebbe essere indipendente da casi custom.
Implementerei una logica per la quale, se vengono passati title e messaggio dell'errore, allora viene mostrato quello altrimenti si mostrano i messaggi di errore di default .

Vediamo nel caso specifico del 400 in fase di creazione di una nuova delega:
- createAppError = (error: { response: { status: number } }) diventerebbe createAppError = (error: { response: { status: number; customMessage?: {title: string; message: string}} })
- va tolto l'if response status === 400 && error... === 'Delega già presente'
- va aggiunto l'if che se esiste customMessage e esistono title e message, di fare il return con quel title e quel message
- nel file actions.ts delle deleghe, l'action createDelegation va modificata nel ramo del catch in questo modo:

if (e.response.status === 400) {
return rejectWithValue({...e.response, customMessage: {title: 'Delega già presente', message: 'La persona che hai indicato ha già una delega per questo ente.'}});
}
return rejectWithValue(e);

in questo modo la gestione di messaggi di errori custom viene data alle singole api e evita che per ogni messaggio custom si debba aggiungere un if nel message service

implementare una logica simile per l'errore 500

Reply:
fatte le modifiche richieste